compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -118,6 +121,10 @@ TargetAttr createTargetAttr(const TargetDetails &details, StringRef arch,
 
 //===----------------------------------------------------------------------===//
 // Known AMD target details
+//
+// Note: the max workgroup size is given as signed int32 max because MLIR's
```

Comment:
What's the limit reported by `rocm-smi`? (Although I think `rocm-smi` reported number doesn't reflect the reality iirc?) We can make the field as int64 or optional if necessary.

Reply:
While `rocm-smi` does report `0xff_ff_ff_ff`, that's a problem in MLIR, because we use `index` for the workgroup ID, which is signed.

(I'm also not sure if that number is in units of workgroups or workitems, so it might be `0xff_ff_ff_ff / work_group_size`)

Reply:
Hm, yeah, I'd be open to doing that

Reply:
let's do it as a follow up later. not blocking.